diff -ruN cldr_tools.orig/org/unicode/cldr/posix/POSIX_LCMonetary.java cldr_tools/org/unicode/cldr/posix/POSIX_LCMonetary.java
--- cldr_tools.orig/org/unicode/cldr/posix/POSIX_LCMonetary.java	2009-02-03 17:52:56.270149335 +0100
+++ cldr_tools/org/unicode/cldr/posix/POSIX_LCMonetary.java	2009-02-03 17:51:34.635545327 +0100
@@ -146,6 +146,9 @@
    int rightparen_position = monetary_formats[POSITIVE].indexOf(')',decimal_symbol_position);
    int space_position = monetary_formats[POSITIVE].indexOf(' ');
 
+   if (space_position == -1)
+      space_position = monetary_formats[POSITIVE].indexOf('\u00a0');
+
    if ( currency_symbol_position >= 0) 
       if ( currency_symbol_position > decimal_symbol_position )
          p_cs_precedes = "0";
@@ -217,6 +220,9 @@
    rightparen_position = monetary_formats[NEGATIVE].indexOf(')',decimal_symbol_position);
    space_position = monetary_formats[NEGATIVE].indexOf(' ');
 
+   if (space_position == -1)
+      space_position = monetary_formats[NEGATIVE].indexOf('\u00a0');
+
    if ( currency_symbol_position >= 0) 
       if ( currency_symbol_position > decimal_symbol_position )
          n_cs_precedes = "0";
