diff -ruN cldr_tools.orig/org/unicode/cldr/util/SupplementalDataInfo.java cldr_tools/org/unicode/cldr/util/SupplementalDataInfo.java
--- cldr_tools.orig/org/unicode/cldr/util/SupplementalDataInfo.java	2009-03-24 16:15:07.701136536 +0100
+++ cldr_tools/org/unicode/cldr/util/SupplementalDataInfo.java	2009-03-24 16:16:30.327418739 +0100
@@ -1426,7 +1426,7 @@
 
         // add fractional examples
         final double fraction = (sample + 0.31d);
-        Count fracType = Count.valueOf(pluralRules.select(fraction));
+        Count fracType = Count.valueOf(pluralRules.select(sample));
         boolean addCurrentFractionalExample = false;
         
         if (fracType == Count.other) {
@@ -1512,7 +1512,7 @@
     }
 
     public Count getCount(double exampleCount) {
-      return Count.valueOf(pluralRules.select(exampleCount));
+      return Count.valueOf(pluralRules.select((long)exampleCount));
     }
 
     public String getRules() {
