diff -ruN cldr_tools.orig/org/unicode/cldr/posix/GenerateCharmap.java cldr_tools/org/unicode/cldr/posix/GenerateCharmap.java
--- cldr_tools.orig/org/unicode/cldr/posix/GenerateCharmap.java	2009-03-24 19:42:32.436441698 +0100
+++ cldr_tools/org/unicode/cldr/posix/GenerateCharmap.java	2009-03-24 19:44:26.787467519 +0100
@@ -85,7 +85,7 @@
     
     public GenerateCharmap(UnicodeSet chars, Charset cs , String codeset) {
         this.cs = cs;
-        if (cs != null) {
+        if (cs != null && ! cs.name().equals("UTF-8")) {
             UnicodeSet csset = new SimpleConverter(cs).getCharset();
         	chars = new UnicodeSet(chars).retainAll(csset);
         }
