diff -ruN cldr_tools.orig/org/unicode/cldr/posix/POSIX_LCMessages.java cldr_tools/org/unicode/cldr/posix/POSIX_LCMessages.java
--- cldr_tools.orig/org/unicode/cldr/posix/POSIX_LCMessages.java	2009-04-17 15:24:51.025762080 +0200
+++ cldr_tools/org/unicode/cldr/posix/POSIX_LCMessages.java	2009-04-17 15:45:33.653943613 +0200
@@ -41,7 +41,7 @@
          if ( n != null && (( s = LDMLUtilities.getNodeValue(n)) != null ))
          {
             StringBuffer buf;
-            if ( variant.yesno.equals("short"))
+            if ( variant.yesno.equals("short") || variant.yesno.equals("solaris"))
             {
                i = s.indexOf(":");
                if ( i > 0 )
@@ -58,9 +58,13 @@
                  buf.append(":yes:y:YES:Y");
             }
             yesstr = POSIXUtilities.POSIXCharNameNP(buf.toString());
+	    
+            if (variant.yesno.equals("solaris") && ! s.startsWith("yes")) {
+                 buf.insert(0, "yes:");
+            }
             yesexpr = POSIXUtilities.POSIXYesNoExpr(buf.toString());
          }
-         else if ( variant.yesno.equals("short"))
+         else if ( variant.yesno.equals("short") || variant.yesno.equals("solaris"))
          {
             yesstr = "yes";
             yesexpr = POSIXUtilities.POSIXYesNoExpr(yesstr);
@@ -76,7 +80,7 @@
          if ( n != null && (( s = LDMLUtilities.getNodeValue(n)) != null ))
          {
             StringBuffer buf;
-            if ( variant.yesno.equals("short"))
+            if ( variant.yesno.equals("short") || variant.yesno.equals("solaris"))
             {
                i = s.indexOf(":");
                if ( i > 0 )
@@ -93,9 +97,13 @@
                  buf.append(":no:n:NO:N");
             }
             nostr = POSIXUtilities.POSIXCharNameNP(buf.toString());
+            
+	    if (variant.yesno.equals("solaris") && ! s.startsWith("no")) {
+                 buf.insert(0, "no:");
+            }
             noexpr = POSIXUtilities.POSIXYesNoExpr(buf.toString());
          }
-         else if ( variant.yesno.equals("short"))
+         else if ( variant.yesno.equals("short") || variant.yesno.equals("solaris"))
          {
             nostr = "no";
             noexpr = POSIXUtilities.POSIXYesNoExpr(nostr);
diff -ruN cldr_tools.orig/org/unicode/cldr/posix/POSIXUtilities.java cldr_tools/org/unicode/cldr/posix/POSIXUtilities.java
--- cldr_tools.orig/org/unicode/cldr/posix/POSIXUtilities.java	2009-04-17 15:24:51.027292298 +0200
+++ cldr_tools/org/unicode/cldr/posix/POSIXUtilities.java	2009-04-17 15:48:56.418081710 +0200
@@ -527,8 +527,11 @@
          {   
             if ( result.length() > 0 )
                result.append(")|(");
-            else
-               result.append("^((");
+            else {
+               result.append("^(");
+	       if (YesNoElements.length > 1) 
+		  result.append("(");
+            }
 
             StringCharacterIterator si = new StringCharacterIterator(cur);
             boolean OptLastChars = false;
@@ -553,7 +556,9 @@
                result.append(")?");
          }
       }
-      result.append("))");
+      result.append(")");
+      if (YesNoElements.length > 1) 
+	 result.append(")");
       return(POSIXCharNameNP(result.toString()));
    }
 }
