diff -ruN cldr_tools.orig/org/unicode/cldr/posix/POSIX_LCCollate.java cldr_tools/org/unicode/cldr/posix/POSIX_LCCollate.java
--- cldr_tools.orig/org/unicode/cldr/posix/POSIX_LCCollate.java
+++ cldr_tools/org/unicode/cldr/posix/POSIX_LCCollate.java
@@ -91,6 +91,7 @@
         }
      }
 
+     col.setStrength(RuleBasedCollator.IDENTICAL);
      allItems = new SortedBag(col);
      contractions = new SortedBag(col);
 
